src/Parser/JsonChecker.cs

```diff
@@ -0,0 +1,441 @@
+#region License, Terms and Author(s)
+//
```

Comment:
So `JsonChecker` is required because otherwise the following compliance test:

https://github.com/jdevillard/JmesPath.Net/blob/1f47b842ade8271aa181b8f86be90e2a4200194d/tools/jmespathnet.compliance/tests/syntax.json#L543-L547

fails with:

    $ dotnet run -p .\tools\jmespathnet.compliance -t .\tools\jmespathnet.compliance\tests
    …
    Executing compliance test syntax.json...
    …
    foo[?bar==`["foo`bar"]`]...Evaluation failed ; Expected error: syntax, but completed without errors.
    …
    Compliance summary:
    Success rate: 99.89%, 870/871 succeeded, 1/871 failed.

There seems to be the expectation (and rightly so) that an invalid literal should stall parsing of the remainder of the expression. This is why it belongs in this PR.